docs/DifferentiableFunctions.md

```diff
@@ -0,0 +1,444 @@
+# Differentiable functions and differentiation APIs
+
+[Richard Wei], [Dan Zheng], [Marc Rasi], [Parker Schuh]
+
+Last updated: March 2019
+
+## Introduction
+
+Swift supports differentiable functions as part of the language. The
+`@differentiable` attribute appears in two locations in Swift syntax: as an
+annotation on function types and as an annotation on function declarations (and
+other similar declarations). This document explains the meaning of these
+annotations.
+
+## `@differentiable` function type attribute
+
+### Basics
+
+In Swift, function types can have attributes. When a function type is annotated
+with `@differentiable`, Swift guarantees that all values of that function type
+can be differentiated.
+
+`@differentiable` functions can be called like normal functions, or be passed to
+APIs that take `@differentiable` functions, like [`gradient(of:)`]. The binary
```

Comment:
The `gradient(of:)` seems to be linking to this 
https://tensorflow.devsite.corp.google.com/swift/api_docs/Functions#/s:10TensorFlow8gradient2of15CotangentVectorQzxcAA0A0Vyq_Gxc_tAA14DifferentiableRzAA0aB13FloatingPointR_r0_lF 
which seems inaccessible

Reply:
This is true for the links to the differentiation APIs table as well

Reply:
Thanks for the catch! Fixed.